### ISSUE TITLE
Training on custom dataset (accuracy skyrocket after 20 iter ) 

### ISSUE THREAD

**Issue by bendidi** [facebookresearch/Detectron#104]
### Expected results

a normal training procedure , on custom data that I transformed to coco format 

### Actual results

the results are as follow : 

the accuracy_cls increased from 0.006 to 0.981445 in 20 iterations ?? and after testing on test set I get 0 mAP  

`json_stats: {"accuracy_cls": 0.006836, "eta": "24 days, 23:45:01", "iter": 0, "loss": 5.010650, "loss_bbox": 0.009897, "loss_cls": 4.306221, "loss_rpn_bbox_fpn2": 0.000000, "loss_rpn_bbox_fpn3": 0.000000, "loss_rpn_bbox_fpn4": 0.000000, "loss_rpn_bbox_fpn5": 0.000000, "loss_rpn_bbox_fpn6": 0.001996, "loss_rpn_cls_fpn2": 0.524622, "loss_rpn_cls_fpn3": 0.139630, "loss_rpn_cls_fpn4": 0.022800, "loss_rpn_cls_fpn5": 0.000000, "loss_rpn_cls_fpn6": 0.005484, "lr": 0.003333, "mb_qsize": 64, "mem": 4698, "time": 5.397753}`

`json_stats: {"accuracy_cls": 0.981445, "eta": "4 days, 18:54:52", "iter": 20, "loss": 0.405047, "loss_bbox": 0.046638, "loss_cls": 0.198343, "loss_rpn_bbox_fpn2": 0.000000, "loss_rpn_bbox_fpn3": 0.000000, "loss_rpn_bbox_fpn4": 0.000000, "loss_rpn_bbox_fpn5": 0.000000, "loss_rpn_bbox_fpn6": 0.008189, "loss_rpn_cls_fpn2": 0.005927, "loss_rpn_cls_fpn3": 0.017690, "loss_rpn_cls_fpn4": 0.014823, "loss_rpn_cls_fpn5": 0.003969, "loss_rpn_cls_fpn6": 0.011517, "lr": 0.003600, "mb_qsize": 64, "mem": 4718, "time": 1.034282}`

`json_stats: {"accuracy_cls": 0.974121, "eta": "3 days, 18:41:34", "iter": 40, "loss": 0.239387, "loss_bbox": 0.061252, "loss_cls": 0.156821, "loss_rpn_bbox_fpn2": 0.000000, "loss_rpn_bbox_fpn3": 0.000000, "loss_rpn_bbox_fpn4": 0.000000, "loss_rpn_bbox_fpn5": 0.000000, "loss_rpn_bbox_fpn6": 0.004920, "loss_rpn_cls_fpn2": 0.000000, "loss_rpn_cls_fpn3": 0.000023, "loss_rpn_cls_fpn4": 0.002601, "loss_rpn_cls_fpn5": 0.004215, "loss_rpn_cls_fpn6": 0.009396, "lr": 0.003867, "mb_qsize": 64, "mem": 4718, "time": 0.816319}`

### Detailed steps to reproduce

The config file I used is : `e2e_faster_rcnn_X-101-64x4d-FPN_2x.yaml`
the command I used is  :  `python tools/train_net.py --multi-gpu-testing --cfg configs/12_2017_baselines/e2e_faster_rcnn_X-101-64x4d-FPN_2x.yaml OUTPUT_DIR RUNS/Test_v1
`
data is custom , but in the exact same format as coco with 77 classes in and a fixed image size of 800x900

### System information

* Operating system: ubuntu 16.04
* CUDA version: 8
* cuDNN version: 6
* NVIDIA driver version: 384
* GPU models (for all devices if they are not all the same):  x2 GTX 1080 
* `python --version` output: 2.7

**Comment by dmasmont** [comment]
@bendidi I have a similar problem, high accuracy_cls and low mAP results. 
How did you solve the problem?

**Comment by RafaRuiz** [comment]
@bendidi @dmasmont any explanation?

**Comment by dmasmont** [comment]
Hello @RafaRuiz , I think the classification accuracy is related on the number of rois used at training (TRAIN.BATCH_SIZE_PER_IM). If you select 1, accuracy cls will probably be 100% (as It will only use background rois) but your final mAP will be extremely low, as the network will only have learned to classify background. The best way to evaluate the network is to use a validation set and compute the mAP on it.
